pkg/probe/util_test.go

```diff
@@ -41,3 +42,105 @@ func TestFindPortByName(t *testing.T) {
 		t.Errorf("Expected %v, got %v, err: %v", want, got, err)
 	}
 }
+
+func TestResolveContainerPort(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		param     intstr.IntOrString
+		container *v1.Container
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    int
+		wantErr bool
+	}{
+		{
+			name: "get port by int type",
+			args: args{
+				param:     intstr.IntOrString{Type: 0, IntVal: 443},
+				container: &v1.Container{},
+			},
+			want:    443,
+			wantErr: false,
+		},
+		{
+			name: "invalid port",
+			args: args{
+				param:     intstr.IntOrString{Type: 0, IntVal: 66666},
+				container: &v1.Container{},
+			},
+			want:    66666,
+			wantErr: true,
+		},
+		{
+			name: "get port by port name",
+			args: args{
+				param: intstr.IntOrString{Type: 1, StrVal: "foo"},
+				container: &v1.Container{
+					Ports: []v1.ContainerPort{
+						{
+							Name:          "foo",
+							ContainerPort: 8080,
+						},
+						{
+							Name:          "bar",
+							ContainerPort: 9000,
+						},
+					},
+				},
+			},
+			want:    8080,
+			wantErr: false,
+		},
+		{
+			name: "no port name",
+			args: args{
+				param: intstr.IntOrString{Type: 1, StrVal: "foo"},
+				container: &v1.Container{
+					Ports: []v1.ContainerPort{
+						{
+							Name:          "bar",
+							ContainerPort: 9000,
+						},
+					},
+				},
+			},
+			want:    0,
+			wantErr: true,
+		},
+		{
+			name: "invalid param type",
+			args: args{
+				param: intstr.IntOrString{Type: 2, StrVal: "foo"},
+				container: &v1.Container{
+					Ports: []v1.ContainerPort{
+						{
+							Name:          "foo",
+							ContainerPort: 8080,
+						},
+						{
+							Name:          "bar",
+							ContainerPort: 9000,
+						},
+					},
+				},
+			},
+			want:    -1,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
```

Comment:
```suggestion
	for _, tt := range tests {
		tt := tt	
```

Another nit, I think you should copy the `tt` var here to avoid a race when accessing it from the subtests running i parallel. You can eg. see [here](https://github.com/kubernetes/kubernetes/commit/6d85f947bf86d12e4459d31029c41d6049d40c0e).

Reply:
Seems like the `pull-kubernetes-verify` is complaining about the same @my-git9 😄 

```
# k8s.io/kubernetes/pkg/probe
pkg/probe/util_test.go:136:37: loop variable tt captured by func literal
pkg/probe/util_test.go:136:52: loop variable tt captured by func literal
pkg/probe/util_test.go:137:23: loop variable tt captured by func literal
pkg/probe/util_test.go:138:68: loop variable tt captured by func literal
pkg/probe/util_test.go:141:14: loop variable tt captured by func literal
pkg/probe/util_test.go:142:59: loop variable tt captured by func literal
make[1]: *** [Makefile:339: vet] Error 
```

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/115708/pull-kubernetes-verify/1625149019476987904